Enhance documentation on layouting GLSP diagrams (#56)
eclipsesource/graphical-lsp-website: content/documentation/rendering/content.md

Note that the `SLabelView` also checks whether the given element is visible and skips the SVG generation if the element is not visible in the diagram canvas.
This check is optional but it’s highly recommended to implement it in your custom views as it heavily improves the rendering performance.

The following sections give an overview of available default views in Sprotty and GLSP and how to configure them:

#### Default Sprotty Views

##### CircularNodeView

A `CircularNodeView` creates a round shape with a radius computed from the shape's size (by default it computes the radius by the minimum of the shape's width or height and divides that by 2).

> The computation of the radius can be overriden and adapted to custom needs.

Reviewer: Do you mean with "overriden" that it can be overriden by specifying the value in the GModelElement? I think this should be specified more clearly.

Author: Adressed in PR #57

```ts

> configureModelElement(

Reviewer: I'm not sure how useful the registration code is. I think it'd be more useful (but also more work), to show an example of creating such a node in the GModelFactory and show a screenshot of the result.

Author: Adressed in PR https://github.com/eclipsesource/graphical-lsp-website/pull/57

  context,
  DefaultTypes.NODE_CIRCLE,
  CircularNode,
  CircularNodeView
);
```

##### DiamondNodeView

A `DiamondNodeView` creates a rhombus shape based on the shape's size.

```ts
configureModelElement(
  context,
  DefaultTypes.NODE_DIAMOND,
  DiamondNode,
  DiamondNodeView
);
```

##### ExpandButtonView

> The `ExpandButtonView` renders a SVG element in the shape of a triangle that allows exapandable parent elements to trigger expansion, for example to display further element information.

Reviewer: ```suggestion
The `ExpandButtonView` renders a SVG element in the shape of a triangle that allows expandable parent elements to trigger expansion, for example to display further element information.
```

```ts
configureModelElement(
  context,
  DefaultTypes.BUTTON_EXPAND,
  SButton,
  ExpandButtonView
);
```

##### PreRenderedView

> The `PreRenderedView` visualizes a previously rendered piece of svg code as a separate SVG element.

Reviewer: ```suggestion
The `PreRenderedView` visualizes a previously rendered piece of svg code as a separate SVG element.
This enables putting SVG code directly in the graphical model, which may be useful for including complex images for certain use cases.
However, usually it is recommended to create a dedicated element type and register a dedicated view, which produces custom SVG, as this yields more flexibility to take bounds, etc., into account.
```

```ts
configureModelElement(
  context,
  DefaultTypes.PRE_RENDERED,
  PreRenderedElement | ShapedPreRenderedElement,
  PreRenderedView
);
```

##### RectangularNodeView

> A `RectangularNodeView` creates a rectangular shape based shape's size.

Reviewer: ```suggestion
A `RectangularNodeView` creates a rectangular shape based on the shape's size.
```

```ts
configureModelElement(
  context,
  DefaultTypes.NODE_RECTANGLE,
  RectangularNode,
  RectangularNodeView
);
```

##### SGraphView

> The `SGraphView` renders the base svg canvas for an SModel and triggers the rendering of its children.

Reviewer: ```suggestion
The `SGraphView` renders the base SVG canvas for an SModel and triggers the rendering of its children.
```

```ts
configureModelElement(context, DefaultTypes.GRAPH, GLSPGraph, SGraphView);
```

##### SLabelView

The `SLabelView` renders a text element that contains the given label text.

```ts
configureModelElement(context, DefaultTypes.LABEL, SLabel, SLabelView);
```

##### SRoutingHandleView

A `SRoutingHandleView` renders a circle shaped element that servers as routing point for routable elements (e.g. Edges).

> Its position is computed by either a register `EdgeRouterRegistry` or passed routing arguments.

Reviewer: ```suggestion
Its position is computed either by a registered `EdgeRouterRegistry` or the routing arguments of the element.
```

```ts
configureModelElement(
  context,
  DefaultTypes.ROUTING_POINT,
  SRoutingHandle,
  SRoutingHandleView
);
```

#### Default GLSP Views

##### GEdgeView

A `GEdgeView` renders a line element which is routed by the `EdgeRouterRegistry`.
The view also triggers the rendering of additional elements (such as mouse handles) and edge children (such as edge labels or routing points).

```ts
configureModelElement(context, DefaultTypes.EDGE, SEdge, GEdgeView);
```

##### GIssueMarkerView

> A `GIssueMarkerView` renders a issue marker on top of shapes that have a validation issue.

Reviewer: ```suggestion
A `GIssueMarkerView` renders an issue marker on top of shapes. This is used to show validation results on elements (see Model Validation).
```

These issue markers are elements in the shape of an information, warning or error icon based on the severity of the issue.

```ts
configureModelElement(
  context,
  DefaultTypes.ISSUE_MARKER,
  SIssueMarker,
  GIssueMarkerView
);
```

##### RoundedCornerNodeView

> A `RectangularNodeView` creates a rectangular shape based shape's size and computes and renders the corners in a rounded way, based on the given options (i.e. `getClipPathInsets`).

Reviewer: I think the corner radius defaults to 0, so it is worth mentioning that the corner radius needs to be specified.
Also I'd mention all arguments and how they need to be passed.

This is related to the comment above that an example of creating a GModelElement that uses this type might be more helpful than showing the registration.

Also a link to the source code might be very helpful in general for all the views mentioned here.

Author: Thanks, yes I totally agree, I addressed this in PR https://github.com/eclipsesource/graphical-lsp-website/pull/57

```ts
configureModelElement(context, DefaultTypes.NODE, SNode, RoundedCornerNodeView);
```

##### StructureCompartmentView

> 

Reviewer: What is that?

Author: Oh sorry, I oversaw that section, I addressed this in PR https://github.com/eclipsesource/graphical-lsp-website/pull/57

```ts
configureModelElement(
  context,
  "struct",
  SCompartment,
  StructureCompartmentView
);
```

##### ForeignObjectView

> The `ForeignObjectView` renders a box , which is usually contained by a default node view, which takes care of resizing and moving of the element.

Reviewer: ```suggestion
The `ForeignObjectView` renders elements that are foreign to SVG, such as HTML, MathML, etc. as specified in their `namespace` and `code` property. Usually such an element is contained by a node view that enables features, such as resizing and moving of the element.
```

> The view then renders the provided code snippet, e.g. a `<pre>` element.

Reviewer: ```suggestion
```

```ts
configureModelElement(
  context,
  DefaultTypes.FOREIGN_OBJECT,
  ForeignObjectElement,
  ForeignObjectView,
  {
    disable: [selectFeature, moveFeature],
  }
);
```

> An example use case for using a `ForeignObjectView` is for example a multiline-text-box.

Reviewer: ```suggestion
A common example use case for using a `ForeignObjectView` is to benefit from word wrapping support of HTML to show multiline text box.
```
